tests/features/composite-keys/GH5629.test.ts

```diff
@@ -193,3 +193,31 @@ test('GH issue 5629, createCompositeKeyArray', async () => {
   }));
   expect(compositeKeys).toMatchSnapshot();
 });
+
+
+test.failing(`GH issue 5629, query fields`, async () => {
+  const tenant = orm.em.create(Tenant, { id: '2' });
+  const something = orm.em.create(Something, { tenant, id: '3' });
+  const x1 = orm.em.create(SomethingThatBelongsToSomething, { tenant, something, id: '4' });
+  const x2 = orm.em.create(SomethingThatBelongsX2, { tenant, something, x1, id: '5' });
+  const x3 = orm.em.create(SomethingThatBelongsX3, { tenant, something, x1, x2, id: '6' });
+  orm.em.create(SomethingThatBelongsX4, { tenant, something, x1, x2, x3 });
+
+  await orm.em.flush();
+
+  const resolved = (await orm.em.findOne(SomethingThatBelongsX4, {
+    tenant,
+    something,
+    x1,
+    x2,
+    x3,
+  }))!;
+
+  expect(resolved).toMatchObject({
+    tenant,
+    something,
+    x1,
+    x2,
+    x3,
+  });
```

Comment:
@B4nan any tips on where to look to fix this one? the issue boils down to the primaryKey hash for an entity having no uniques (e.g. tenantId~~~somethingId~~~x1Id) while the array used for the where clause duplicates it, generating the following error:

```select `s0`.* from `something_that_belongs_x4` as `s0` where `s0`.`tenant_id` = '2' and (`s0`.`tenant_id`, `s0`.`something_id`) in ( values ('2', '3')) and (`s0`.`tenant_id`, `s0`.`something_id`, `s0`.`x1_id`) in ( values ('2', '2', '3', '4')) and (`s0`.`tenant_id`, `s0`.`something_id`, `s0`.`x1_id`, `s0`.`x2_id`) in ( values ('2', '2', '3', '2', '2', '3', '4', '5')) and (`s0`.`tenant_id`, `s0`.`something_id`, `s0`.`x1_id`, `s0`.`x2_id`, `s0`.`x3_id`) in ( values ('2', '2', '3', '2', '2', '3', '4', '2', '2', '3', '2', '2', '3', '4', '5', '6')) limit 1 - SQLITE_ERROR: sub-select returns 16 columns - expected 5
DriverException: select `s0`.* from `something_that_belongs_x4` as `s0` where `s0`.`tenant_id` = '2' and (`s0`.`tenant_id`, `s0`.`something_id`) in ( values ('2', '3')) and (`s0`.`tenant_id`, `s0`.`something_id`, `s0`.`x1_id`) in ( values ('2', '2', '3', '4')) and (`s0`.`tenant_id`, `s0`.`something_id`, `s0`.`x1_id`, `s0`.`x2_id`) in ( values ('2', '2', '3', '2', '2', '3', '4', '5')) and (`s0`.`tenant_id`, `s0`.`something_id`, `s0`.`x1_id`, `s0`.`x2_id`, `s0`.`x3_id`) in ( values ('2', '2', '3', '2', '2', '3', '4', '2', '2', '3', '2', '2', '3', '4', '5', '6')) limit 1 - SQLITE_ERROR: sub-select returns 16 columns - expected 5```

can't simply remove duplicates in QueryHelper, because there are times where you want a duplicate (E.g. tenantId = 1 or tenantId = 2)

I suspect that patching this in findOne won't be sufficient, because there are still other places where this could work (e.g. find all the x3's belonging to a specific x2)